processor/src/bitwise/mod.rs

```diff
@@ -117,6 +117,13 @@ impl Bitwise {
         self.trace[0].len()
     }
 
+    /// Returns the starting row address of the most recently executed bitwise operation cycle. It
+    /// expects at least one bitwise operation to have been executed already.
+    pub fn last_cycle_start_row(&self) -> Felt {
+        debug_assert!(self.trace_len() >= OP_CYCLE_LEN, "bitwise trace too short");
+        Felt::new((self.trace_len() - OP_CYCLE_LEN) as u64)
+    }
```

Comment:
Maybe instead of last cycle start row address here we can return it together with the result from the individual methods - i.e., `u32and`, `u32or`, and `u32xor`.

Also, if I remember correctly, in the latest design of bitwise constraints we are actually interested in the last row of the cycle rather than the first one.

Reply:
> Maybe instead of last cycle start row address here we can return it together with the result from the individual methods - i.e., u32and, u32or, and u32xor.

Yep, this works for me.

> Also, if I remember correctly, in the latest design of bitwise constraints we are actually interested in the last row of the cycle rather than the first one.

I'll check. I had based this on the u32 design doc.

Reply:
I left a comment in the u32 design doc so that we can correct the documentation when it's migrated. Please let me know if I made an error in that comment.

Reply:
The comment looks good! Thank you!